include/cache.h

```diff
@@ -13,59 +13,185 @@
 extern "C" {
 #endif
 
-void arch_dcache_flush(void *addr, size_t size);
-void arch_dcache_invd(void *addr, size_t size);
+/**
+ * Common operations for the caches
+ */
+#define K_CACHE_WB	BIT(0)
+#define K_CACHE_INVD	BIT(1)
+#define K_CACHE_WB_INVD	(K_CACHE_WB | K_CACHE_INVD)
+
+/**
+ *
+ * @brief Enable d-cache
+ *
+ * Enable the d-cache.
+ *
+ * @return N/A
+ */
+void arch_dcache_enable(void);
+
+/**
+ *
+ * @brief Disable d-cache
+ *
+ * Disable the d-cache.
+ *
+ * @return N/A
+ */
+void arch_dcache_disable(void);
+
+/**
+ *
+ * @brief Enable i-cache
+ *
+ * Enable the i-cache.
+ *
+ * @return N/A
+ */
+void arch_icache_enable(void);
+
+/**
+ *
+ * @brief Disable i-cache
+ *
+ * Disable the i-cache.
+ *
+ * @return N/A
+ */
+void arch_icache_disable(void);
+
+/**
+ *
+ * @brief Write-back / Invalidate / Write-back + Invalidate one level d-cache
+ *
+ * Write-back, Invalidate or Write-back + Invalidate one lever of the d-cache
+ *
+ * @param level	The cache level to act on
+ * @param op	Operation to perform (one of the K_CACHE_* operations)
+ *
+ * @retval 0		On success
+ * @retval -ENOTSUP	If the operation is not supported
```

Comment:
Does _"the operation is not supported"_ include the case when `K_CACHE_WB_INVD` is attempted on a cache which supports the `INVD` but not the `WB`? In other words, if the application needs to know the cache capabilities.

Reply:
I didn't attach any particular meaning to that but your interpretation matches mine. If the arch returns `-ENOTSUP` that means that that particular operation is not supported so yes, the application can use that to probe the cache capabilities.